pkgs/development/compilers/llvm/common/compiler-rt/default.nix

```diff
@@ -256,6 +279,11 @@ stdenv.mkDerivation {
     ++ lib.optionals (noSanitizers && lib.versionAtLeast release_version "19") [
       "-DCOMPILER_RT_BUILD_CTX_PROFILE=OFF"
     ]
+    ++ lib.optionals withAtomics (
+      [ (lib.cmakeBool "COMPILER_RT_EXCLUDE_ATOMIC_BUILTIN" (!withAtomicsLib)) ]
+      ++ lib.optional withAtomicsLib (lib.cmakeBool "COMPILER_RT_BUILD_STANDALONE_LIBATOMIC" true)
+      ++ lib.optional withAtomicsPthread (lib.cmakeBool "COMPILER_RT_LIBATOMIC_USE_PTHREAD" true)
```

Comment:
```suggestion
      [
        (lib.cmakeBool "COMPILER_RT_EXCLUDE_ATOMIC_BUILTIN" (!withAtomicsLib))
        (lib.cmakeBool "COMPILER_RT_BUILD_STANDALONE_LIBATOMIC" withAtomicsLib)
        (lib.cmakeBool "COMPILER_RT_LIBATOMIC_USE_PTHREAD" withAtomicsPthread)
      ]
```

Reply:
that's how it was in the previous version (https://github.com/NixOS/nixpkgs/commit/0d7159de41b4505ba31a4640ce049b7be4ea4f3f). I changed it to avoid passing unknown CMake options to old LLVM versions that don't support them

Reply:
We always could use `lib.versionAtLeast`. With the new refactoring, we tried to get less optionals here.

Reply:
Should be fine for at least `COMPILER_RT_BUILD_STANDALONE_LIBATOMIC` since only 1 thing uses LLVM 12

Reply:
LLVM 12 is no more in nixpkgs, and only 18 does not support `COMPILER_RT_LIBATOMIC_USE_PTHREAD`. Passing that option unconditionally should not affect the build output, so using the simpler code makes sense to me.